doc/tor-genkey.1.txt

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) The Tor Project, Inc.
+// See LICENSE for licensing information
+// This is an asciidoc file used to generate the manpage/html reference.
+// Learn asciidoc on http://www.methods.co.nz/asciidoc/userguide.html
+:man source:   Tor
+:man manual:   Tor Manual
+tor-genkey(1)
+==============
+Suphanat Chunhapanya
+
+NAME
+----
+tor-genkey - utility to generate various keys for tor
+
+SYNOPSIS
+--------
+**tor-genkey** [-t|--type __keytype__] [-o|--out __filename__]
+
+DESCRIPTION
+-----------
+**tor-genkey** is a key generation utility to be used to generate various keys
+for many subsystems of tor.
+
+OPTIONS
+-------
+**-t**, **--type** __keytype__::
+    Specify which key type you want to generate.
+
+**-o**, **--output** __filename__::
+    Output your key in __filename__.
+
+KEY TYPES
+---------
+**HsClientDescriptorAuth**::
```

Comment:
I propose we use `HS` instead of `Hs`.

Small fix, maybe: `HSDescriptorClientAuth` or `HSClientAuthDescriptor instead? Since this is "client authorization", we should just namespace it clearly so it is easy to understand/categorized. If we had let say the introduction authorization, then we would have also something like `HSClientAuthIntro` which starts the same way as the Descriptor one.

For usability, I would make it case insensitive in the implementation so probably also good to mention it in the manpage.